src/prefect/environments/execution/local.py

```diff
@@ -14,13 +15,34 @@ class LocalEnvironment(Environment):
     used, the environment variables from this process will be passed.
 
     Args:
+        - executor (Executor, optional): the executor to run the flow with. If not provided, the
+            default executor will be used.
         - labels (List[str], optional): a list of labels, which are arbitrary string identifiers used by Prefect
             Agents when polling for work
         - on_start (Callable, optional): a function callback which will be called before the flow begins to run
         - on_exit (Callable, optional): a function callback which will be called after the flow finishes its run
         - metadata (dict, optional): extra metadata to be set and serialized on this environment
     """
 
+    def __init__(
+        self,
+        executor: "prefect.engine.executors.Executor" = None,
+        labels: Iterable[str] = None,
+        on_start: Callable = None,
+        on_exit: Callable = None,
+        metadata: dict = None,
+    ) -> None:
+        if executor is None:
+            executor = prefect.engine.get_default_executor_class()()
```

Comment:
I think we should only call this method at runtime if an executor is not already attached to the environment (we don't inject local configuration into deployed flows anywhere else and I don't think we should start).  This would ensure local settings aren't surprisingly baked into deployments - instead it allows for them to be updated / changed at the agent level through config.

Reply:
Not disputing (could go either way on this) but this does match the behavior of the existing `RemoteEnvironment` - if no environment is specified the default executor I'm the registration environment is the one that is used.

Reply:
Oh wow, interesting - I still think we should change that at some point but it's probably outside the scope of this PR in that case 👍